src/cmd/build.rs

```diff
@@ -1173,21 +1200,36 @@ mod tests_ci_only {
     #[test]
     fn keep_debug_symbols_in_release_mode() {
         with_new_contract_project(|manifest_path| {
-            let res = super::execute(
-                &manifest_path,
-                Verbosity::Default,
-                BuildMode::Release,
-                BuildArtifacts::CodeOnly,
-                UnstableFlags::default(),
-                OptimizationPasses::default(),
-                true,
-            )
-            .expect("build failed");
+            let args = crate::cmd::build::ExecuteArgs {
+                manifest_path,
+                build_mode: BuildMode::Release,
+                build_artifact: BuildArtifacts::CodeOnly,
+                keep_debug_symbols: true,
+                ..Default::default()
+            };
+
+            let res = super::execute(args).expect("build failed");
 
             // we specified that debug symbols should be kept
             assert!(has_debug_symbols(&res.dest_wasm.unwrap()));
 
             Ok(())
         })
     }
+
+    #[test]
+    fn build_result_seralization_sanity_check() {
+        with_new_contract_project(|manifest_path| {
+            let args = crate::cmd::build::ExecuteArgs {
+                manifest_path,
+                output_type: OutputType::Json,
+                ..Default::default()
+            };
+
+            let res = super::execute(args).expect("build failed");
+
+            assert!(res.serialize_json().is_ok());
```

Comment:
```suggestion
            // then
            assert!(res.serialize_json().is_ok());
```

Reply:
Would appreciate deserializing here and checking if the fields are in there (as you wrote above).

Reply:
I think it would be best to add a 
```
let expected output = /* Insert a multi-line string of the whole fixture */;
assert_eq!(serialized_json, expected_output);
```
You will have to replace some part of the absolute artifact paths in the serialized json (or use a regex).

Reply:
Could you add this test back in? It ran the whole execution pipeline with `OutputType::Json`, which is not tested otherwise.